File: salonmanager/backend/app/Models/CustomerProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Support\Tenancy\SalonOwned;

class CustomerProfile extends Model
{
    use SalonOwned;

    protected $fillable = ['salon_id', 'user_id', 'phone', 'preferred_stylist', 'prefs', 'address'];
    protected $casts = ['prefs' => 'array', 'address' => 'array'];

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: salonmanager/frontend/lib/features/customer_profile/models/customer_profile.dart
class CustomerProfile {
  final String? phone;
  final String? preferredStylist;
  final Map<String, dynamic> prefs;
  final Map<String, dynamic> address;

  CustomerProfile({
    this.phone,
    this.preferredStylist,
    this.prefs = const {},
    this.address = const {},
  });

  factory CustomerProfile.fromJson(Map<String, dynamic> j) => CustomerProfile(
        phone: j['phone'],
        preferredStylist: j['preferred_stylist'],
        prefs: Map<String, dynamic>.from(j['prefs'] ?? {}),
        address: Map<String, dynamic>.from(j['address'] ?? {}),
      );
}